bin/publish

```diff
@@ -1,15 +1,67 @@
-#!/bin/bash -e
-#
-# Push CyberArk sidecar injector mutating webhook service
-# usage: ./bin/publish
-set -ex
+#!/bin/bash
 
-if [ -z "$1" ]
-then
-    echo "tag not provided";
-    exit 1;
-fi
+set -e
 
-cd $(dirname $0)/..
+cd "$(dirname "${0}")"
+. ./build_utils
+cd ..
 
-docker push cyberark/sidecar-injector:$1
+readonly IMAGE_NAME="cyberark/sidecar-injector"
+readonly IMAGE_WITH_TAG="${IMAGE_NAME}:$(git_tag)"
+
+function publish_image() {
+  local tag=${1}
+  image_with_version_tag="${IMAGE_NAME}:${tag}"
+  echo "Tagging and pushing ${image_with_version_tag}"
+
+  docker tag "${IMAGE_WITH_TAG}" "${image_with_version_tag}"
+  docker push "${image_with_version_tag}"
+}
+
+function main() {
+  # If an argument is passed to the script it will be read as a tag, and this single tag
+  # will be published then the script will exit. This functionality is useful for
+  # publishing single tags without involving the tag-triggered builds logic below.
+  if [[ $# -gt 0 ]]
+  then
+      publish_image $1
+      exit 0;
+  fi
+
+  if [[ -z "${TAG_NAME:-}" ]]; then
```

Comment:
Wouldn't:
```bash
if [[ -z "$TAG_NAME" ]]; then
```
be equivalent here?

Reply:
`set -u` will complain if `TAG_NAME` is altogether not defined.